src/libraries/Native/Unix/System.Security.Cryptography.Native/pal_ssl.c

```diff
@@ -41,6 +46,24 @@ void CryptoNative_EnsureLibSslInitialized()
 #elif OPENSSL_VERSION_NUMBER < OPENSSL_VERSION_1_1_0_RTM
     EnsureLibSsl10Initialized();
 #endif
+
+    // Get length of session master key and generate one.
+    SSL_CTX* ctx = SSL_CTX_new(TLS_method());
+    if (ctx != NULL)
+    {
+        s_sessionTicketsMasterKeyLength = SSL_CTX_ctrl(ctx, SSL_CTRL_SET_TLSEXT_TICKET_KEYS, 0, NULL);
+        if (s_sessionTicketsMasterKeyLength > 0)
+        {
+            s_sessionTicketsMasterKey = malloc((size_t)s_sessionTicketsMasterKeyLength);
```

Comment:
What happens if the malloc fails and returns NULL?

Reply:
Ideally, this shouldn't even be using `malloc()` here.  If these keys are leaked somehow, due to the lack of forward secrecy in TLS 1.2, it should be possible to recover the plaintext of resumed sessions.

Since creating servers isn't usually in the fast path of many applications, this could be instead living inside a private, anonymous memory mapping that's marked with `PROT_NONE` to cause a fault when trying to read it.  (It can be `mprotect()`ed to have `PROT_READ` only in the tiny window of time in `CryptoNative_SslCtxCreate()` where that information is needed.)

In addition, it should be advised as `MADV_DONTDUMP` so it doesn't end up in core dumps and whatnot, and mapped with `MAP_LOCKED` to prevent this from being paged out to disk.

Reply:
I can look into if. But would we already the same problem with private keys loaded into generic memory and full handshake using that key @lpereira? As far as I can tell, SSL_CTX_new() from OpenSSL does not seems to do anything special.

Reply:
It's a similar problem, yes, but with performance on the line, it's much harder to solve. The private keys are not only needed for every handshake, but, more importantly, on a potentially busy server, the window where the private keys would be unreadable would be smaller than the time they would need to be readable due to a large number of incoming connections, rendering this kind of protection useless.

For these master keys in particular, the `mprotect()` dance is fine as it's less likely for servers to be spawned, so the price paid for the two syscalls every time a new TLS server is spawned is quickly amortized.

Reply:
this is called for every session (as we do not share SSL_CTX).  So we would need to do unproctect/protect for every handshake, right @lpereira? Furthermore, each SSL_CTX object will have same value.

Reply:
Ah, yeah.  You're right.  Can't use the `PROT_READ`/`PROT_NONE` thing, but at least marking it as `PROT_READ` (so it's never overwritten by mistake or malice), and the other things (don't dump/wipe on fork, locked) would help.  This could be used for the private keys, too.